test: `test_array_dunder_with_dtype` fails in the "random version" ci job

```
__________ test_array_dunder_with_dtype[pandas_nullable_constructor] ___________

request = <FixtureRequest for <Function test_array_dunder_with_dtype[pandas_nullable_constructor]>>
constructor_eager = <function pandas_nullable_constructor at 0x7f35b669e5e0>

    def test_array_dunder_with_dtype(request: Any, constructor_eager: Any) -> None:
        if "pyarrow_table" in str(constructor_eager) and parse_version(
            pa.__version__
        ) < parse_version("16.0.0"):  # pragma: no cover
            request.applymarker(pytest.mark.xfail)
    
        df = nw.from_native(constructor_eager({"a": [1, 2, 3]}), eager_only=True)
>       result = df.__array__(object)

tests/frame/array_dunder_test.py:32: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
narwhals/dataframe.py:237: in __array__
    return self._compliant_frame.__array__(dtype, copy=copy)
narwhals/_pandas_like/dataframe.py:105: in __array__
    return self.to_numpy(dtype=dtype, copy=copy)
narwhals/_pandas_like/dataframe.py:531: in to_numpy
    return self._native_frame.to_numpy(dtype=dtype, copy=copy)
/opt/hostedtoolcache/Python/3.9.19/x64/lib/python3.9/site-packages/pandas/core/frame.py:1993: in to_numpy
    result = self._mgr.as_array(dtype=dtype, copy=copy, na_value=na_value)
/opt/hostedtoolcache/Python/3.9.19/x64/lib/python3.9/site-packages/pandas/core/internals/managers.py:1680: in as_array
    arr = blk.values.to_numpy(  # type: ignore[union-attr]
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <IntegerArray>
[1, 2, 3]
Length: 3, dtype: Int64, dtype = dtype('O')
copy = None, na_value = <NA>

    def to_numpy(
        self,
        dtype: npt.DTypeLike | None = None,
        copy: bool = False,
        na_value: object = lib.no_default,
    ) -> np.ndarray:
        """
        Convert to a NumPy Array.
    
        By default converts to an object-dtype NumPy array. Specify the `dtype` and
        `na_value` keywords to customize the conversion.
    
        Parameters
        ----------
        dtype : dtype, default object
            The numpy dtype to convert to.
        copy : bool, default False
            Whether to ensure that the returned value is a not a view on
            the array. Note that ``copy=False`` does not *ensure* that
            ``to_numpy()`` is no-copy. Rather, ``copy=True`` ensure that
            a copy is made, even if not strictly necessary. This is typically
            only possible when no missing values are present and `dtype`
            is the equivalent numpy dtype.
        na_value : scalar, optional
             Scalar missing value indicator to use in numpy array. Defaults
             to the native missing value indicator of this array (pd.NA).
    
        Returns
        -------
        numpy.ndarray
    
        Examples
        --------
        An object-dtype is the default result
    
        >>> a = pd.array([True, False, pd.NA], dtype="boolean")
        >>> a.to_numpy()
        array([True, False, <NA>], dtype=object)
    
        When no missing values are present, an equivalent dtype can be used.
    
        >>> pd.array([True, False], dtype="boolean").to_numpy(dtype="bool")
        array([ True, False])
        >>> pd.array([1, 2], dtype="Int64").to_numpy("int64")
        array([1, 2])
    
        However, requesting such dtype will raise a ValueError if
        missing values are present and the default missing value :attr:`NA`
        is used.
    
        >>> a = pd.array([True, False, pd.NA], dtype="boolean")
        >>> a
        <BooleanArray>
        [True, False, <NA>]
        Length: 3, dtype: boolean
    
        >>> a.to_numpy(dtype="bool")
        Traceback (most recent call last):
        ...
        ValueError: cannot convert to bool numpy array in presence of missing values
    
        Specify a valid `na_value` instead
    
        >>> a.to_numpy(dtype="bool", na_value=False)
        array([ True, False, False])
        """
        hasna = self._hasna
        dtype, na_value = to_numpy_dtype_inference(self, dtype, na_value, hasna)
        if dtype is None:
            dtype = object
    
        if hasna:
            if (
                dtype != object
                and not is_string_dtype(dtype)
                and na_value is libmissing.NA
            ):
                raise ValueError(
                    f"cannot convert to '{dtype}'-dtype NumPy array "
                    "with missing values. Specify an appropriate 'na_value' "
                    "for this dtype."
                )
            # don't pass copy to astype -> always need a copy since we are mutating
            with warnings.catch_warnings():
                warnings.filterwarnings("ignore", category=RuntimeWarning)
                data = self._data.astype(dtype)
            data[self._mask] = na_value
        else:
            with warnings.catch_warnings():
                warnings.filterwarnings("ignore", category=RuntimeWarning)
>               data = self._data.astype(dtype, copy=copy)
E               ValueError: NoneType copy mode not allowed.

/opt/hostedtoolcache/Python/3.9.19/x64/lib/python3.9/site-packages/pandas/core/arrays/masked.py:526: ValueError

```


versions:
```
attrs==24.2.0
cachetools==5.5.0
cfgv==3.4.0
chardet==5.2.0
click==8.1.7
cloudpickle==3.0.0
colorama==0.4.6
covdefaults==2.3.0
coverage==7.6.1
dask==2024.8.0
dask-expr==1.1.10
distlib==0.3.8
exceptiongroup==1.2.2
filelock==3.15.4
fsspec==2024.6.1
hypothesis==6.111.1
identify==2.6.0
importlib-metadata==8.4.0
iniconfig==2.0.0
joblib==1.4.2
locket==1.0.0
nodeenv==1.9.1
numpy==1.26.4
packaging==24.1
pandas==2.2.2
partd==1.4.2
pip==24.2
platformdirs==4.2.2
pluggy==1.5.0
polars==0.20.16
pre-commit==3.8.0
pyarrow==15.0.0
pyproject-api==1.7.1
pytest==8.3.2
pytest-cov==5.0.0
pytest-env==1.1.3
python-dateutil==2.9.0.post0
pytz==2024.1
pyyaml==6.0.2
setuptools==73.0.1
six==1.16.0
sortedcontainers==2.4.0
threadpoolctl==3.5.0
tomli==2.0.1
toolz==0.12.1
tox==4.18.0
tzdata==2024.1
virtualenv==20.26.3
zipp==3.20.0
```

--- a/narwhals/_pandas_like/dataframe.py
+++ b/narwhals/_pandas_like/dataframe.py
@@ -527,6 +527,10 @@ class PandasLikeDataFrame:
     def to_numpy(self, dtype: Any = None, copy: bool | None = None) -> Any:
         from narwhals._pandas_like.series import PANDAS_TO_NUMPY_DTYPE_MISSING
 
+        if copy is None:
+            # pandas default differs from Polars
+            copy = False
+
         if dtype is not None:
             return self._native_frame.to_numpy(dtype=dtype, copy=copy)
 

